scripts/ci/check_compliance.py

```diff
@@ -863,7 +863,8 @@ def check_no_undef_outside_kconfig(self, kconf):
                 sym_name = sym_name[7:]  # Strip CONFIG_
                 if sym_name not in defined_syms and \
                    sym_name not in self.UNDEF_KCONFIG_ALLOWLIST and \
-                   not (sym_name.endswith("_MODULE") and sym_name[:-7] in defined_syms):
+                   not (sym_name.endswith("_MODULE") and sym_name[:-7] in defined_syms) and \
+                   not (sym_name.endswith("_MODULE_BLOBS") and sym_name[:-13] in defined_syms):
```

Comment:
This is not fixing what you hope to fix, because this is intended for Kconfig tristate settings, where a symbol can take the value: `Y | N | M` (yes, no, module), and where choosing `M` will result in a internally defined symbol named `<setting>_MODULE`. Therefore `<setting>` is in `defined_syms`.

Looking at a solution which works for modules without Kconfig files, but also a solution which must work when running KconfigBasicNoModules check, and ideally can catch module typos in addition, like `ZEPHYR_HLA_TS_MODULE_BLOBS`.

Reply:
Ok. I've removed this change from the PR now.